src/charm.py

```diff
@@ -563,13 +575,9 @@ def _configure(self, _):
 
             logger.info("Prometheus configuration reloaded")
 
-        if (
-            isinstance(self.unit.status, BlockedStatus)
-            and self.unit.status not in early_return_statuses.values()
-        ):
-            return
-
-        self.unit.status = ActiveStatus()
+        self.unit.status = StatusBase._get_highest_priority(
+            cast(List[StatusBase], list(self.status.values()))
+        )
```

Comment:
@benhoyt, just drawing your attention to this status-related (ab)use, if you have a moment to take a look.

Currently there's a mix between centralized and decentralized status setting, and collect-status can't be used without major refactoring.

An alternative could be using the collect-status machinery to commit all values inside `self.status`.

Reply:
Yeah, that sounds like a good direction to me too.

Additionally, I'd prefer charms didn't use private (underscore-prefixed) Ops methods like `StatusBase._get_highest_priority`. That said, I think that one's been useful enough for people doing various things with status that maybe we should make it public, either as `StatusBase.get_highest_priority` or another name. If you're interested in that, please open an issue for us to consider.